### PR TITLE
Cap size before block gossiping

### PR DESCRIPTION
**Pull request by Jason-Zhangxin-Chen**
This PR port an improvement from https://github.com/clearmatics/autonity-internal/pull/272 to this public repo, it aims at limiting the fault scope. Cap the block size to align with the limits of dev-p2p protocol.

### CODE REVIEW

File: miner/worker.go [hunk 1]
```diff
@@ -771,6 +778,14 @@ func (w *worker) commitTransactions(env *environment, txs *types.TransactionsByP
 		if tx == nil {
 			break
 		}
+
+		// Cap the maximum size of packed TXNs in a block before we run into the unexpected errors due to
+		// the limits of msg size in dev-p2p layer. As current ENV is not always exist in legacy tests, check nil to prevent panic.
+		if w.current != nil && w.current.txsSize+tx.Size() > capBlockSize {
```
Comment (lorenzo-dev1) [inline miner/worker.go:784]: I think we should check the `env` parameter, as that is what is passed to `commitTransaction` for adding the txs. Not sure if it always coincide with the `w.current`. @vpiyush

**Comment by yazzaoui** [comment]
To be honest I'm not so sure about this being needed.. we can always cap the block gas limit and that would achieve indirectly the same goal..

**Comment by lorenzo-dev1** [comment]
I agree, I had not thought about controlling it this way.

**Comment by Jason-Zhangxin-Chen** [comment]
Close it now as the team decide to control in in gas limit.

**Pull request closed without merging**